src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -176,6 +201,8 @@ public TaskInfo createTask(Offer offer, TaskID taskId, NodeTask nodeTask) {
             Objects.requireNonNull(offer, "Offer should be non-null");
             Objects.requireNonNull(nodeTask, "NodeTask should be non-null");
 
+            NMPorts ports = getPorts(offer);
+
```

Comment:
Let's add a LOGGER.debug that prints out all the random ports assigned for each NM's end point. Perhaps we should just a toString() method inside NMPorts.java and do ```java LOGGER.debug(nmPorts.toString)```.